### PR TITLE
[ghc] added v9.10.1

### PR DESCRIPTION
**Pull request by timin**
Signed-off-by: Nimit [nimit.jyotiana@hotmail.com](mailto:nimit.jyotiana@hotmail.com)

### CODE REVIEW

**Comment by smandaka** [comment]
@timin  Getting below error message when checking ghc version.  But getting version correctly.

/hab/pkgs/core/ghc/9.10.1/20241010123808/bin$ ./ghc --version
**./ghc: /hab/pkgs/core/ghc/9.10.1/20241010123808/lib/x86_64-linux-ghc-9.10.1/libtinfo.so.6: no version information available (required by /hab/pkgs/core/ghc/9.10.1/20241010123808/lib/x86_64-linux-ghc-9.10.1/libHShaskeline-0.8.2.1-dba2-ghc9.10.1.so)
./ghc: /hab/pkgs/core/ghc/9.10.1/20241010123808/lib/x86_64-linux-ghc-9.10.1/libtinfo.so.6: no version information available (required by /hab/pkgs/core/ghc/9.10.1/20241010123808/lib/x86_64-linux-ghc-9.10.1/libHSterminfo-0.4.1.6-53ae-ghc9.10.1.so)**
The Glorious Glasgow Haskell Compilation System, version 9.10.1

File: packages/development/compilers/ghc/plan.sh [hunk 1]
```diff
@@ -47,9 +47,9 @@ do_install() {
 
 	# The ghc binary (ghc-${pkg_version}) requires libtinfo.so.6, which is not available
 	# in the current ncurses build because it was built with the --enable-widec option.
-	ln -sfv "$(pkg_path_for ncurses)/lib/libtinfow.so" "${pkg_prefix}/lib/x86_64-linux-ghc-${pkg_version}/libtinfo.so.6"
+	ln -sfv "$(pkg_path_for ncurses)/lib/libtinfow.so" "${pkg_prefix}/bin/libtinfo.so.6"
```
Comment (sajjaphani) [inline packages/development/compilers/ghc/plan.sh:50]: The symlink creation should be moved to the ncurses package.

Comment (timin) [reply packages/development/compilers/ghc/plan.sh:50]: I have tried that as well but it did not worked for ghc package. Technically if library path is available linker should pick it up for required symbols. Currently I have included symlink in bin directory of ghc package.

I have tested using "hab pkg exec core/ghc -- ghc --version" which is working without below error

Comment (sajjaphani) [reply packages/development/compilers/ghc/plan.sh:50]: What do you mean by 'not working'? Can you provide more details? It seems odd to me to place the library file (link) in the bin folder and then include it in LD_RUN_PATH.

**Review by smandaka — APPROVED** [review]

File: packages/development/compilers/ghc/plan.sh (hunk 1 above)
Comment (timin) [reply packages/development/compilers/ghc/plan.sh:50]: The issue is posted above by Sangameshwar which is runtime error and I did changes to mitigate that error. For which you have posted suggestion. This has been already been tested and does not resolve the error so suggestion is "not working". Could you share that you have tried or build? may be plan file, I can also build & test your changes to verify!

yes it is odd. GNU/GCC linker has default standard settings for library paths, where library should be present like "/usr/lib", "/lib" etc one of them is searching in current directory where binary resides which is "bin" directory this case. So when linker search for symbols used binaries, it looks that all standard paths for library if anything which is non-standard are included and used by env variables like LD_LIBRARY_PATH, PATH, LD_RUN_PATH etc and others compilation and linking options & arguments with build commands. In this, library present in "bin" works and mitigate error where as library path "<pkg_curses_path>/lib" does not work.

**Comment by smandaka** [comment]
As discused in Binary bakers group, I have approved PR.

**Pull request merged**